### DISCUSSION TITLE
Integral along the boundary

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#25150] [Q&A Modules: General]
Hello,
![image](https://github.com/idaholab/moose/assets/130671073/75f4549d-c907-4e1c-98f0-0a3fcd18d698)
I have an integral that looks like this. Some of the variables in the integral term vary with time and position. I want to store the result of this integral as a material property`[_qp]` because it's location related. Or I want to store such integral values at each integral point of the boundary. How can I do to achieve the results I want in MOOSE?
Thanks.

**Comment by GiudGiud** [comment]
Hello

You can create a material property to do this. 
Is your mesh 1D? 
Is this integral from the bottom of the mesh to the current location in the element?

Are q2, GP, D, F all variables or material properties or functions?

Guillaume

**Reply by jin0123456** [reply]
My mesh is 2D(RZ).
The integral is from the bottom to the current location in the element and I need to store this integral value in every locations(every elem) on boundary.
q2 is material property, GP, D are constant values, F is a function or it can also be a material property.
Thanks.

**Reply by GiudGiud** [reply]
édit: first you will need to form the intégrand using a ParsedAux

Actually you could try with an auxiliary variable and with a LayeredIntegrzal user object and a SpatialUserOvjectAux to use the user object to fill the variable

then in the LayeredIntegrzl you turn on the cumulative option 

this could work for now. You can work with the order of the variable and the number of layers to get a high resolution

**Reply by jin0123456** [reply]
Thanks for your answer,  I have solved the problem in this way.
